arangod/Aql/SimpleModifier.cpp

```diff
@@ -192,8 +195,53 @@ ExecutionState SimpleModifier<ModifierCompletion, Enable>::transact(
   // operations blocking as in previous versions of ArangoDB.
   // TODO: fix this and make it truly non-blocking (requires to
   // fix some lifecycle issues for AQL queries first).
-  result.wait();
 
+  // If we simply wait, it can happen that we get into a blockage in which
+  // all threads wait in the same place here and none can make progress,
+  // since the scheduler is full. This means we must detach the thread
+  // after some time. To avoid that all are detaching at the same time,
+  // we choose a random timeout for the detaching. But first we spin a
+  // while to avoid delays:
+  if (!result.isReady()) {
+    auto const spinTime = std::chrono::microseconds(10'000);
```

Comment:
I would suggest to use milliseconds here to make it more intuitive.
```suggestion
    auto const spinTime = std::chrono::milliseconds(10);
```

Reply:
Done.